### ISSUE TITLE
[Feature Request] Option to employ a lighthouse as bounce server

### ISSUE THREAD

**Issue by nfam** [slackhq/nebula#144]
UDP hole punching can get peers connected in most cases. However, when we carry our laptops from place to place, a sophisticated NAT or network device between may stop them from establishing the connection to nodes through UDP hole punching. We need a backup solution, since, at the moment, performance is not our concern, connecting to specific nodes and get the job done is the priority.

Let's say, an option in `config.yml` of a specific `node A` (mostly a laptop) to tell a lighthouse that "I choose you as my bounce server". Other nodes can get that message from the lighthouse, and any connection from and to `node A` will go through the hop lighthouse.

When `node A` gets back to the safe place, i.e. bring the laptop back to our company office from the customer office, just edit the config to turn off the option. Then direct connection by UDP hole punching will be employed again.

In short, the feature is not meant to be always on, it is a backup solution, but a critical one, which guarantees that we can always use nebula at any location.

**Comment by rawdigits** [comment]
I've actually spec'd this out as well, but it requires some changes to operation that we probably won't get to for a while it isn't quite as easy as telling traffic to go through the lighthouse, since every header in a nebula packet only contains an index number relating to the remote host, and no other identifying info. It is do-able, but I don't know if/when we will do it. It requires changes to lighthouse+nebula in not subtle ways.

It also requires a signaling mechanism where nodes can request forwarding of an "index number" to a remote IP/port on their behalf. This also has security considerations.

Your idea isn't far off from what I wrote out, with the exception that we need to handle things like DoS/etc and prevent relaying of packets from arbitrary hosts/etc. It has utility for sure.

**Comment by nfam** [comment]
I realise that I can connect my laptop to the lighthouse via WireGuard, or to any VPN server which NAT does not block the way, then use this connection for Nebula by setting `local_range`. The performance is not that good, but it can work.

Personally, if this feature requires significant design change that may affect performance and introduce some kind of risk, I suggest **not to implement it.**

I close this for now.
